drivers/flash/flash_gecko.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2018, Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <kernel.h>
+#include <device.h>
+#include <em_msc.h>
+#include <flash.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_gecko);
+
+struct flash_gecko_data {
+	struct k_sem mutex;
```

Comment:
call it lock or else, but not mutex as we have struct k_mutex type. To avoid confusion